### DISCUSSION TITLE
Gas Well Forecast: Decline Curve and Production Rate

### DISCUSSION THREAD

**Discussion by j-craggy** [has2k1/plotnine#795] [2024 Plotnine Contest]
### Authors

Jeff Craggy

### Links

[ShinyApps.io Submission](https://kraggle.shinyapps.io/jcplotnine/)

### Full description

Because my plot can have a few different formats, I decided to give Shiny for Python a shot, and make this an interactive project. The data set it ficticous natural gas well production and forecast model with quality metrics. The plot uses historic well production data for the model fit then the forecast was generated about 2 years into production. Finally we can see the quality of the forecast how it held up over time. These metrics are useful to stakeholders in determining future production rates of their energy assets. 

- Decline Rate / Forecasts
- Cumulative estimated ultimate recovery 

Both plots are available in normal and log scale curves with optional quality report. The code was written in R Studio then in VS Code for Python deployment to ShinyApps.io which can be found here: [Shiny for Python](https://shiny.posit.co/py/docs/overview.html) and here: [Shinyapps.io](https://www.shinyapps.io/)

Enjoy!

### Code repository

Codebase: [GitHub](https://github.com/JBC-Inc/jbcPlotnine)
Submission: [ShinyApps.io](https://kraggle.shinyapps.io/jcplotnine/)

**Comment by has2k1** [comment]
<img width="1395" alt="Screenshot 2024-07-29 at 13 49 26" src="https://github.com/user-attachments/assets/3ad4605b-f8db-4af0-bd70-037636cd130b">
<hr/>

<img width="1395" alt="Screenshot 2024-07-29 at 13 49 33" src="https://github.com/user-attachments/assets/ec5ea9e7-ffed-427a-a76a-155ca018305d">
<hr/>

<img width="1395" alt="Screenshot 2024-07-29 at 13 49 47" src="https://github.com/user-attachments/assets/0520811f-4d59-47dd-b673-cd8e3300435a">
<hr/>

<img width="1395" alt="Screenshot 2024-07-29 at 13 49 59" src="https://github.com/user-attachments/assets/098bff9d-2775-4522-b6a5-2c99ca7d7ed4">

**Comment by has2k1** [comment]
@j-craggy, The winning submission has been announced [here](https://posit.co/blog/winner-of-the-2024-plotnine-plotting-contest/). Thank you for taking part in the contest.
